packages/nimble-components/src/table/components/row/template.ts

```diff
@@ -2,13 +2,16 @@ import { html, repeat } from '@microsoft/fast-element';
 import { DesignSystem } from '@microsoft/fast-foundation';
 import type { TableRow } from '.';
 import { TableCell } from '../cell';
+import type { TableColumn } from '../../../table-column/base';
 
 // prettier-ignore
 export const template = html<TableRow>`
     <template role="row">
-        ${repeat(x => x.columns, html<string>`
+        ${repeat(x => x.columns, html<TableColumn>`
```

Comment:
you can specify the parent context in the type so we can avoid the `as TableRow` below:

```suggestion
        ${repeat(x => x.columns, html<TableColumn, TableRow>`
```

Similar to:

https://github.com/ni/nimble/blob/6804c569e9d1a10cb79f1a799eb9be194ddc42f7/packages/nimble-components/src/breadcrumb/tests/breadcrumb.stories.ts#L55-L59